OpenSSL_version.mk

```
@@ -1,80 +1,107 @@
+            # $(OPENSSL_DIR) is absolute path
+            OPENSSL_LIB = $(OPENSSL_DIR)/$(LIB)
+        endif
+        $(info [TRACE] After OPENSSL_DIR check: OPENSSL_LIB="$(OPENSSL_LIB)")
+        CFLAGS += -isystem $(OPENSSL_DIR)/include
+        LDFLAGS += -L$(OPENSSL_DIR) -L$(OPENSSL_LIB) -Wl,-rpath,$(OPENSSL_DIR) -Wl,-rpath,$(OPENSSL_LIB)
+    endif
+
+    LDLIBS = -lcrypto
 
-.phony: default build show clean
+    $(info [TRACE] After OPENSSL_DIR set: EXE=$(EXE))
+    $(info [TRACE] After OPENSSL_DIR set: LIB=$(LIB))
+    $(info [TRACE] After OPENSSL_DIR set: OPENSSL_LIB=$(OPENSSL_LIB))
+    $(info [TRACE] After OPENSSL_DIR set: CFLAGS=$(CFLAGS))
+    $(info [TRACE] After OPENSSL_DIR set: LDFLAGS=$(LDFLAGS))
+    $(info [TRACE] After OPENSSL_DIR set: LDLIBS=$(LDLIBS))
 
-default: build show clean
+    .PHONY: default build show clean
 
-build: OpenSSL_version
+    default: build show clean
 
-show:	build
+    build: OpenSSL_version
+
+    show: build
 	@./OpenSSL_version$(EXE)
 
-clean:
+    clean:
 	@rm -f OpenSSL_version$(EXE)
 
-
 else ifeq ($(LIB),header)
+    $(info [DEBUG] LIB is set to header)
 
+    OPENSSL_NUMBER_SEL = head -n 1 | sed -r 's/.*OpenSSL //' | awk '{print ($$0+0)}'
+    OPENSSLV_H = $(OPENSSL_DIR)/include/openssl/opensslv.h
+    ifeq ($(shell fgrep OPENSSL_VERSION_MAJOR "$(OPENSSLV_H)"),)
+        $(info [DEBUG] OPENSSL_VERSION_MAJOR not found in OPENSSLV_H)
+        OPENSSL_VERSION = $(shell grep 'OPENSSL_VERSION_TEXT\s* "OpenSSL ' "$(OPENSSLV_H)" | $(OPENSSL_NUMBER_SEL))
+    else
+        $(info [DEBUG] OPENSSL_VERSION_MAJOR found in OPENSSLV_H)
+        ifeq ($(OS),MacOS)
+            $(info [DEBUG] OS is MacOS)
+            OPENSSL_VERSION = $(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $4"."$8 }')
+        else
+            $(info [DEBUG] OS is not MacOS)
+            OPENSSL_VERSION = $(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $$4"."$$8 }')
+        endif
+        $(info [TRACE] After OS check in header: OPENSSL_VERSION=$(OPENSSL_VERSION))
+    endif
 
-OPENSSL_NUMBER_SEL=head -n 1 | sed -r 's/.*OpenSSL //' | awk '{print ($$0+0)}'
-OPENSSLV_H=$(OPENSSL_DIR)/include/openssl/opensslv.h
-ifeq ($(shell fgrep OPENSSL_VERSION_MAJOR "$(OPENSSLV_H)"),)
-  OPENSSL_VERSION=$(shell grep 'OPENSSL_VERSION_TEXT\s* "OpenSSL ' "$(OPENSSLV_H)" | $(OPENSSL_NUMBER_SEL))
-else
-  ifeq ($(OS),MacOS)
-    OPENSSL_VERSION=$(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $4"."$8 }')
-  else
-    OPENSSL_VERSION=$(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $$4"."$$8 }')
-  endif
-endif
-
-ifeq ($(OPENSSL_VERSION),1)
-    OPENSSL_VERSION=1.0
-endif
+    ifeq ($(OPENSSL_VERSION),1)
+        OPENSSL_VERSION = 1.0
+    endif
 
+    $(info [TRACE] OPENSSL_NUMBER_SEL=$(OPENSSL_NUMBER_SEL))
+    $(info [TRACE] OPENSSLV_H=$(OPENSSLV_H))
+    $(info [TRACE] OPENSSL_VERSION=$(OPENSSL_VERSION))
```

I guess it will be sufficient to adapt this line and a similar one below like this:
```suggestion
    $(info [TRACE] OPENSSL VERSION=$(OPENSSL_VERSION))
```